scripts/src/java/org/oppia/android/scripts/regex/RegexPatternValidationCheck.kt

```diff
@@ -0,0 +1,203 @@
+ */
+private fun retrieveFilenameChecks(): List<FilenameCheck> {
+  return getProto(
+    "filename_pattern_validation_checks.pb",
+    FilenameChecks.getDefaultInstance()
+  ).getFilenameChecksList()
+}
+
+/**
+ * Retrieves all file content checks.
+ *
+ * @return a list of all the FileContentChecks
+ */
+private fun retrieveFileContentChecks(): List<FileContentCheck> {
+  return getProto(
+    "file_content_validation_checks.pb",
+    FileContentChecks.getDefaultInstance()
+  ).getFileContentChecksList()
+}
+
+/**
+ * Helper function to parse the textproto file to a proto class.
+ *
+ * @param textProtoFileName name of the textproto file to be parsed
+ * @param proto instance of the proto class
+ * @return proto class from the parsed textproto file
+ */
+private fun <T : MessageLite> getProto(textProtoFileName: String, proto: T): T {
+  val protoBinaryFile = File("scripts/assets/$textProtoFileName")
+  val builder = proto.newBuilderForType()
+
+  // This cast is type-safe since proto guarantees type consistency from mergeFrom(),
+  // and this method is bounded by the generic type T.
+  @Suppress("UNCHECKED_CAST")
+  val protoObj: T =
+    FileInputStream(protoBinaryFile).use {
+      builder.mergeFrom(it)
+    }.build() as T
+  return protoObj
+}
+
+/**
+ * Checks for a prohibited file naming pattern.
+ *
+ * @param repoPath the path of the repo
+ * @param searchFiles a list of all the files which needs to be checked
+ * @param filenameCheck proto object of FilenameCheck
+ * @return file name pattern is correct or not
+ */
+private fun checkProhibitedFileNamePattern(
+  repoPath: String,
+  searchFiles: List<File>,
+  filenameCheck: FilenameCheck,
+): Boolean {
+  val prohibitedFilenameRegex = filenameCheck.getProhibitedFilenameRegex().toRegex()
+
+  val matchedFiles = searchFiles.filter { file ->
+    return@filter file.name !in filenameCheck.getExemptedFileNameList() &&
+      prohibitedFilenameRegex.matches(
+        RepositoryFile.retrieveRelativeFilePath(
+          file,
+          repoPath
+        )
+      )
+  }
+
+  logProhibitedFilenameFailure(filenameCheck.getFailureMessage(), matchedFiles)
+  return matchedFiles.isNotEmpty()
+}
+
+/**
+ * Checks for a prohibited file content.
+ *
+ * @param repoPath the path of the repo
+ * @param searchFiles a list of all the files which needs to be checked
+ * @param fileContentCheck proto object of FileContentCheck
+ * @return file content pattern is correct or not
+ */
+private fun checkProhibitedContent(
+  searchFiles: List<File>,
+  fileContentCheck: FileContentCheck
+): Boolean {
+  val fileNameRegex = fileContentCheck.getFilenameRegex().toRegex()
+
+  val prohibitedContentRegex =
+    fileContentCheck.getProhibitedContentRegex().toRegex()
+
+  val matchedFiles = searchFiles.filter { file ->
+    file.name !in fileContentCheck.getExemptedFileNameList() &&
+      fileNameRegex.matches(file.name) &&
+      File(file.toString()).bufferedReader().lineSequence()
```

Comment:
```suggestion
      File(file.toString())
          .bufferedReader()
          .lineSequence()
```

And then ``.foldIndexed(...)`` would be aligned with ``lineSequence``. This is generally how formatting in these cases should look.

Reply:
Noted.
Now, updated.